libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -82,35 +160,78 @@ void ConstantEvaluator::endVisit(Identifier const& _identifier)
 	ASTPointer<Expression> const& value = variableDeclaration->value();
 	if (!value)
 		return;
-	else if (!m_types->count(value.get()))
+	else if (!evaluated(*value))
 	{
 		if (m_depth > 32)
 			m_errorReporter.fatalTypeError(5210_error, _identifier.location(), "Cyclic constant definition (or maximum recursion depth exhausted).");
-		ConstantEvaluator(m_errorReporter, m_depth + 1, m_types).evaluate(*value);
+
+		evaluate(*value);
 	}
 
-	setType(_identifier, type(*value));
+	// Link LHS's identifier to the evaluation result of the RHS expression.
+	if (auto const resultOpt = result(*value); resultOpt.has_value())
+		setResult(_identifier, TypedValue{variableDeclaration->annotation().type, resultOpt.value().evaluatedValue});
```

Comment:
Where is the actual conversion happening?

Reply:
Good point, not implemented. Should be the `evaluateBinary(lhs, rhs, op)` function I think.

Reply:
Ah, currently I'm only allowing `RationalNumberType` (with annotated type integer) to enter this part, otherwise (currently) the existing code path is executed. I am not sure how often this happens though and how often this does make sense. Because from what I understand so far, the constant evaluator is only requested for code parts that _should_ represent numerical expressions (no string ops, bool? I don't know yet).